collectors/python.d.plugin/mysql/mysql.chart.py

```diff
@@ -612,10 +631,64 @@ def _get_data(self):
             else:
                 self.queries.pop('slave_status')
 
+        if 'user_statistics' in raw_data:
+            if raw_data['user_statistics'][0]:
+                userstats_vars = [e[0] for e in raw_data['user_statistics'][1]]
+                for i in range(len(raw_data['user_statistics'][0])):
+                    name = raw_data['user_statistics'][0][i][0]
+                    userstats_raw_data = dict(zip(userstats_vars, raw_data['user_statistics'][0][i]))
+                    if len(self.charts) > 0:
+                        if ('userstats_{0}_Cpu_time'.format(name)) not in self.charts['userstats_cpu']:
+                            self.add_userstats(name)
+                    for key in USER_STATISTICS:
+                        if key in userstats_raw_data:
+                            to_netdata['userstats_{0}_{1}'.format(name, key)] = userstats_raw_data[key]
+            else:
+                self.queries.pop('user_statistics')
+
         if 'variables' in raw_data:
             variables = dict(raw_data['variables'][0])
             for key in VARIABLES:
                 if key in variables:
                     to_netdata[key] = variables[key]
 
         return to_netdata or None
+
+    def add_userstats(self, name):
+        self.charts['userstats_cpu'].add_dimension(['userstats_{0}_Cpu_time'.format(name), name, 'incremental', 100, 1])
+
+        rows = self.charts.add_chart(self.userstats_rows_chart(name))
+        rows.add_dimension(['userstats_{0}_Rows_read'.format(name), 'read', 'incremental'])
+        rows.add_dimension(['userstats_{0}_Rows_send'.format(name), 'send', 'incremental'])
+        rows.add_dimension(['userstats_{0}_Rows_updated'.format(name), 'updated', 'incremental'])
+        rows.add_dimension(['userstats_{0}_Rows_inserted'.format(name), 'inserted', 'incremental'])
+        rows.add_dimension(['userstats_{0}_Rows_deleted'.format(name), 'deleted', 'incremental'])
+
+        commands = self.charts.add_chart(self.userstats_commands_chart(name))
+        commands.add_dimension(['userstats_{0}_Select_commands'.format(name), 'select', 'incremental'])
+        commands.add_dimension(['userstats_{0}_Update_commands'.format(name), 'update', 'incremental'])
+        commands.add_dimension(['userstats_{0}_Other_commands'.format(name), 'other', 'incremental'])
+
+    @staticmethod
+    def userstats_rows_chart(name):
+        return [
+            'userstats_rows_' + name,
+            None,
+            '{0} User Rows Operations'.format(name.title()),
+            'operations/s',
+            'userstats',
+            'mysql.userstats_rows_' + name,
+            'stacked'
+        ]
+
+    @staticmethod
+    def userstats_commands_chart(name):
+        return [
+            'userstats_commands_' + name,
+            None,
+            '{0} User Commands'.format(name.title()),
+            'commands/s',
+            'userstats',
+            'mysql.userstats_commands_' + name,
+            'stacked'
+        ]
```

Comment:
convert both these methods into functions and move them to the CHARTS (since they are charts too).

Or

I suggest to implement it like this

1. `userstats_charts_template` func

https://github.com/netdata/netdata/blob/master/collectors/python.d.plugin/beanstalk/beanstalk.chart.py#L117-L168

2. `create_new_userstats_charts` method

https://github.com/netdata/netdata/blob/30c1fe15b7c46b9f947b7723dc74497a5198cc4e/collectors/python.d.plugin/beanstalk/beanstalk.chart.py#L217-L226

Reply:
this has been done